js_modules/dagit/packages/core/src/app/QueryRefresh.tsx

```diff
@@ -111,6 +118,35 @@ export function useQueryRefreshAtInterval(queryResult: QueryResult<any, any>, in
   );
 }
 
+/**
+ * This hook allows you to hook a single QueryRefreshCountdown component to more than
+ * one useQueryRefreshAtInterval. The QueryRefreshCountdown will reflect the countdown
+ * state of the FIRST query, but clicking the refresh button will trigger them all.
+ *
+ * Note: If you use this hook, you should pass the same interval to each
+ * useQueryRefreshAtInterval.
+ */
+export function useMergedRefresh(
+  a: QueryRefreshState,
+  b: QueryRefreshState,
+  c?: QueryRefreshState,
+  d?: QueryRefreshState,
+): QueryRefreshState {
+  return React.useMemo(() => {
+    const refetch: ObservableQuery['refetch'] = async () => {
+      const states = [a, b, c, d].filter(Boolean);
+      const [ar] = await Promise.all(states.map((s) => s?.refetch()));
+      return ar!;
+    };
+    return {
+      nextFireMs: a.nextFireMs,
+      nextFireDelay: a.nextFireDelay,
+      networkStatus: a.networkStatus,
+      refetch,
+    };
+  }, [a, b, c, d]);
```

Comment:
You could provide an array (`QueryRefreshState[]`) as the `useMergedRefresh` arg and here, then at the callsite, ensure that you're providing a memoized array.